go/vt/vtgate/planbuilder/ddl.go

```diff
@@ -116,6 +141,17 @@ func buildDDLPlans(sql string, ddlStatement sqlparser.DDLStatement, reservedVars
 		}, nil
 }
 
+func checkFKError(vschema ContextVSchema, ddlStatement sqlparser.DDLStatement) error {
+	if fkStrategyMap[strings.ToLower(vschema.ForeignKeyMode())] == disallow {
```

Comment:
Seems like every call will have to run `ToLower`. Consider moving `ToLower` into `ForeignKeyMode()` func.